tests/conftest.py

```diff
@@ -2976,6 +3017,71 @@ def finalizer():
     return factory
 
 
+@pytest.fixture()
+def multi_snapshot_restore_factory(snapshot_restore_factory):
+    """
+    Snapshot restore factory. Calling this fixture creates set of new PVC out of the
+    specified VolumeSnapshot.
+
+    """
+    instances = []
+
+    def factory(
+        snapshot_obj,
+        restore_pvc_suffix=None,
+        storageclass=None,
+        size=None,
+        volume_mode=None,
+        restore_pvc_yaml=None,
+        access_mode=constants.ACCESS_MODE_RWO,
+        status=constants.STATUS_BOUND,
+        wait_each=False
+    ):
+        """
+        Args:
+            snapshot_obj (list): List OCS instance of kind VolumeSnapshot which has
+                to be restored to new PVC
+            restore_pvc_name_suffix (str): Suffix to be added to pvc name
+            storageclass (str): Name of storageclass
+            size (str): Size of PVC being created. eg: 5Gi. Ideally, this
+                should be same as the restore size of snapshot. Adding this
+                parameter to consider negative test scenarios.
+            volume_mode (str): Volume mode for PVC. This should match the
+                volume mode of parent PVC.
+            restore_pvc_yaml (str): The location of pvc-restore.yaml
+            access_mode (str): This decides the access mode to be used for the
+                PVC. ReadWriteOnce is default.
+            status (str): If provided then factory waits for the PVC to reach
+                desired state.
+            wait_each(bool): True to wait for each PVC to be in status 'status'
+                before creating next PVC, False otherwise
+
+        Returns:
+            PVC: Restored PVC object
+
+        """
+        new_pvcs = []
+        for snap_obj in snapshot_obj:
+            log.info(f"Creating a PVC from snapshot {snap_obj.name}")
+            restore_pvc_name = f"{snap_obj.name}-{restore_pvc_suffix}"\
+                if restore_pvc_suffix else None
+            restored_pvc = snapshot_restore_factory(
+                snapshot_obj=snap_obj, restore_pvc_name=restore_pvc_name,
+                storageclass=storageclass, size=size,
+                volume_mode=volume_mode, restore_pvc_yaml=restore_pvc_yaml,
+                access_mode=access_mode
+            )
+            restored_pvc.snapshot = snapshot_obj
+            new_pvcs.append(restored_pvc)
+            instances.append(restored_pvc)
+            if status and wait_each:
+                helpers.wait_for_resource_state(restored_pvc, status)
```

Comment:
Status will be checked in snapshot_restore_factory because the default value of 'status' in snapshot_restore_factory is 'Bound'.
Pass ```status=status``` to snapshot_restore_factory if wait_each is True.
Check status of all PVCs outside the loop 'for snap_obj in snapshot_obj' if wait_each is False.